tslib/src/functions/gias-sync/gias.service.spec.ts

```diff
@@ -0,0 +1,211 @@
+    })
+
+    sut = new GiasService(
+                soapMessageBuilderMock,
+                soapRequestServiceMock,
+                xmlParserMock,
+                multipartMessageParserMock,
+                zipServiceMock)
+  })
+
+  test('subject should be defined', () => {
+    expect(sut).toBeDefined()
+  })
+
+  test('GetExtract:should propogate original error details when a fault occurs', async () => {
+    const errorInfo = `errorId:${uuid()}`
+    try {
+      soapMessageBuilderMock.buildMessage = jest.fn(() => { throw new Error(errorInfo) })
+      await sut.GetExtract(extractId)
+      fail('error should have been thrown')
+    } catch (error) {
+      expect(error).toBeDefined()
+      expect(error.message).toEqual(errorInfo)
+    }
+  })
+
+  test('GetExtract: soapMessage should reflect extract specification', async () => {
+    let capturedSpecification: ISoapMessageSpecification = {
+      action: '',
+      messageExpiryMs: 0,
+      namespace: ''
+    }
+    soapMessageBuilderMock.buildMessage = jest.fn((messageSpec) => {
+      capturedSpecification = messageSpec
+      return ''
+    })
+    await sut.GetExtract(extractId)
+    expect(capturedSpecification).toBeDefined()
+    expect(capturedSpecification.action).toEqual('GetExtract')
+    expect(capturedSpecification.parameters).toBeDefined()
+    expect(capturedSpecification.parameters['Id']).toEqual(extractId)
+  })
+
+  test('when namespace is not defined an error is thrown', async () => {
+    try {
+      config.Gias.Namespace = undefined
+      await sut.GetExtract(extractId)
+      fail('error was expected to be thrown')
+    } catch (error) {
+      expect(error).toBeDefined()
+      expect(error.message).toEqual('gias web service namespace is required')
+    }
+  })
+
+  test('when serviceUrl is not defined an error is thrown', async () => {
+    try {
+      config.Gias.ServiceUrl = undefined
+      await sut.GetExtract(extractId)
+      fail('error was expected to be thrown')
+    } catch (error) {
+      expect(error).toBeDefined()
+      expect(error.message).toEqual('gias service url is required')
+    }
+  })
+
+  test('when username is not defined an error is thrown', async () => {
+    try {
+      config.Gias.Username = undefined
+      await sut.GetExtract(extractId)
+      fail('error was expected to be thrown')
+    } catch (error) {
+      expect(error).toBeDefined()
+      expect(error.message).toEqual('gias username is required')
+    }
+  })
+
+  test('when password is not defined an error is thrown', async () => {
+    try {
+      config.Gias.Password = undefined
+      await sut.GetExtract(extractId)
+      fail('error was expected to be thrown')
+    } catch (error) {
+      expect(error).toBeDefined()
+      expect(error.message).toEqual('gias password is required')
+    }
+  })
+
+  test.todo('GetExtract: verify returned data structure')
+
+  test.skip('GetExtract:should return an empty object if no results', async () => {
```

Comment:
Why are there 3 skipped tests?

Reply:
they are integration tests that are useful for debugging the real payload.  have added comments to indicate their purpose.  will be removed or moved to suitable location before feature completion.